Enhance SGW & Loyola Maps: Search, User Location, Recenter Button, and UI Improvements

b-chrit opened the pull request:
## **Demo Video**  

https://github.com/user-attachments/assets/01eee0f1-fd5e-4e47-a4de-a04b5de71902


---

### **Summary:**  
This PR improves the **SGW and Loyola maps** by adding **search functionality, user location tracking, a recenter button, and UI enhancements**. Additionally, the **search bar has been removed from the Layout header** and integrated directly into the **SGWMap** and **LoyolaMap** components for better usability.  

It also improves the **BuildingMarker component**, refining the **Callout pop-up UI** for a cleaner and more user-friendly experience.  

Furthermore, **the building colors in both SGW and Loyola maps have been unified** to maintain a **consistent visual identity**, using a deep red (`#9B1B30`) for better visibility and aesthetics.  

### **Key Features:**  

#### **SGW & Loyola Maps Enhancements**  
- **Added Search Functionality:**  
  - Users can now search for buildings directly within **SGWMap** and **LoyolaMap**.  
  - The map will animate to the selected building when searched.  
- **User Location Tracking:**  
  - Integrated `expo-location` to display the user’s location on both maps.  
  - The user is represented with a **red dot inside a white outline** for visibility.  
- **Recenter Button:**  
  - Added a **📍 button** to toggle between the **user’s location** and **SGW/Loyola campus**.  
  - Appears only when the user is **far from the campus** or moves the map away.  
- **Improved Map Region Handling:**  
  - `animateToRegion` is used for **smooth transitions** when searching or recentering.  
- **Refactored `getCentroid` for Buildings:**  
  - Ensured **accurate boundary handling** for computing **building centroids**.  
  - Applied a **coordinate offset for SP building** in Loyola for better positioning.  
- **Unified Building Colors for SGW & Loyola:**  
  - All buildings in **both campuses** now use a **consistent deep red (`#9B1B30`)**.  
  - This enhances **visual consistency and accessibility** on the map.  

#### **BuildingMarker Component Enhancements**  
- **Refined Callout Design:**  
  - The `Get Directions` button is now **centered** inside the callout.  
  - Improved padding and layout for **better visual hierarchy**.  
- **Removed Unnecessary Callout Information:**  
  - Previously displayed **building purpose, facilities, address, and contact** fields were removed for a **cleaner, more focused design**.  

#### **Layout Adjustments**  
- **Removed the Search Bar from the Header:**  
  - The search bar has been moved from the **Layout header** to the **SGW and Loyola maps** instead.  
  - This keeps the layout **cleaner** while improving the **map user experience**.  

---
  
  ### **✅ Closes US 11, US 20, and US 22**

Nabih21 commented:
Great improvements! However, I think that when we click on a button we should have this big popup with all the details, it should something brief that doesn't take space, and maybe having a clickable "see more"  to see the rest of the details ?

Nabih21 commented on FindMyClass/components/SearchBar.js:
```diff
@@ -0,0 +1,36 @@
+import { View, TextInput, StyleSheet } from 'react-native';
+import { Ionicons } from '@expo/vector-icons';
+
+const SearchBar = ({ value, onChangeText, placeholder }) => {
```
For the search bar I recommend using the same logic used in the Destination search bar in the directions page.  We could extract it as a component and use it in the the 2 pages. The search bar i have also shows suggestions while the user is searching, I think this could be better. You can check the implementation in the PR #87

b-chrit commented:
@Nabih21 I have addressed your 1st comment related to reducing the size of the callout by adding 2 buttons to the callout: directions and more info, when clicking on directions it takes you to the directions page and when clicking on more info an alert popup with more information about the building in question, I have opted out to use an alert instead of a custom modal/callout because when attempting to use them, the custom modal when take very long to open and when closing it the map would freeze and it would only be functional again if you were to visit another page and then come back to the map page, we can leave it as an alert for it but we will definitely open a US for it so that we can fix it downline as it has been taking a lot of time.



For the second comment, I have created a SearchBar to reduce code duplication which is being used in both sgw and loyola map and which you can use in your PR as well and customize it as much as possible and i have also added an auto complete feature to it as well

Nabih21 commented:
> For the second comment, I have created a SearchBar component to reduce code duplication which is being used in both sgw and loyola map and which you can use in your PR as well (you can obv customize it as much as possible ) I have also added an auto-complete feature to it as well

Awesome, this looks great and will be more efficient for the users. Thank you for your efforts!
